### PR TITLE
mdatagen: Add Init function for intializing existing metrics

### PR DESCRIPTION
**Pull request by jrcamp**
pdata.Metric instances may already be initialized and we just want to fill in
the metric descriptor values. Removes the unnecessary need to New() then
CopyTo().

Benchmarks

Before

```
BenchmarkCPUScraper-12    	   37053	     29415 ns/op	   12022 B/op	     141 allocs/op
```

After

```
BenchmarkCPUScraper-12    	   45709	     25343 ns/op	   11918 B/op	     138 allocs/op
```

### CODE REVIEW

File: service/internal/resources.go [hunk 1]
```diff
@@ -1,17 +1,3 @@
-// Copyright  The OpenTelemetry Authors
```
Comment (jrcamp) [inline service/internal/resources.go:1]: I think something changed with addlicense. :(

Comment (bogdandrutu) [reply service/internal/resources.go:1]: Yes, addlicense changed to not check/add license for generated code.

File: receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go [hunk 2]
```diff
@@ -27,20 +27,27 @@ const Type configmodels.Type = "hostmetricsreceiver"
...
+	name     string
+	initFunc func(pdata.Metric)
 }
 
 func (m *metricImpl) Name() string {
 	return m.name
 }
 
 func (m *metricImpl) New() pdata.Metric {
-	return m.newFunc()
```
Comment (bogdandrutu) [inline receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go:43]: Maybe we should consider to remove the New (or deprecate it)

**Review by bogdandrutu — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go (hunk 2 above)
Comment (jrcamp) [reply receiver/hostmetricsreceiver/internal/metadata/generated_metrics.go:43]: Hmm, well the New is useful if you're not trying to optimize allocations and just do New/Appends.

**Comment by tigrannajaryan** [comment]
@jrcamp build failed.

**Comment by jrcamp** [comment]
@tigrannajaryan fixed

**Pull request merged**